command/command_testonly/server_testonly_test.go

```diff
@@ -0,0 +1,216 @@
+			},
+		},
+	}
+
+	disabledResponse := &vault.RequestLimiterResponse{
+		GlobalDisabled:   true,
+		ListenerDisabled: false,
+		Limiters: map[string]*vault.LimiterStatus{
+			limits.WriteLimiter: {
+				Enabled: false,
+			},
+			limits.SpecialPathLimiter: {
+				Enabled: false,
+			},
+		},
+	}
+
+	cases := []struct {
+		name        string
+		configAfter string
+		disabled    bool
+	}{
+		{
+			"enable after default",
+			baseHCL + requestLimiterEnableHCL,
+			false,
+		},
+		{
+			"enable after enable",
+			baseHCL + requestLimiterEnableHCL,
+			false,
+		},
+		{
+			"disable after enable",
+			baseHCL + requestLimiterDisableHCL,
+			true,
+		},
+		{
+			"default after disable",
+			baseHCL,
+			false,
+		},
+		{
+			"default after default",
+			baseHCL,
+			false,
+		},
+		{
+			"disable after default",
+			baseHCL + requestLimiterDisableHCL,
+			true,
+		},
+		{
+			"disable after disable",
+			baseHCL + requestLimiterDisableHCL,
+			true,
+		},
+	}
+
+	ui, srv := command.TestServerCommand(t)
+
+	f, err := os.CreateTemp(t.TempDir(), "")
+	require.NoErrorf(t, err, "error creating temp dir: %v", err)
+
+	_, err = f.WriteString(baseHCL)
+	require.NoErrorf(t, err, "cannot write temp file contents")
+
+	configPath := f.Name()
+
+	var output string
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		code := srv.Run([]string{"-config", configPath})
+		output = ui.ErrorWriter.String() + ui.OutputWriter.String()
+		require.Equal(t, 0, code, output)
+	}()
+
+	select {
+	case <-srv.StartedCh():
+	case <-time.After(5 * time.Second):
+		t.Fatalf("timeout")
+	}
+	defer func() {
+		srv.ShutdownCh <- struct{}{}
+		wg.Wait()
+	}()
+
+	err = f.Close()
+	require.NoErrorf(t, err, "unable to close temp file")
+
+	// create a client and unseal vault
+	cli, err := srv.Client()
+	require.NoError(t, err)
+	require.NoError(t, cli.SetAddress("http://127.0.0.1:8209"))
+	initResp, err := cli.Sys().Init(&api.InitRequest{SecretShares: 1, SecretThreshold: 1})
+	require.NoError(t, err)
+	_, err = cli.Sys().Unseal(initResp.Keys[0])
+	require.NoError(t, err)
+	cli.SetToken(initResp.RootToken)
+
+	output = ui.ErrorWriter.String() + ui.OutputWriter.String()
+	require.Contains(t, output, "Request Limiter: enabled")
+
+	verifyLimiters := func(t *testing.T, expectedDisabled bool) {
+		t.Helper()
+		statusResp, err := cli.Logical().Read("/sys/internal/request-limiter/status")
+		require.NoError(t, err)
+		require.NotNil(t, statusResp)
+
+		limitersResp, ok := statusResp.Data["request_limiter"]
+		require.True(t, ok)
+		require.NotNil(t, limitersResp)
+
+		var limiters *vault.RequestLimiterResponse
+		err = mapstructure.Decode(limitersResp, &limiters)
+		require.NoError(t, err)
+		require.NotNil(t, limiters)
+
+		switch expectedDisabled {
+		case true:
+			require.Equal(t, disabledResponse, limiters)
+		default:
+			require.Equal(t, enabledResponse, limiters)
+		}
+	}
+
+	verifyLimiters(t, false)
+
+	for _, tc := range cases {
+		tc := tc
```

Comment:
nit: the subtests aren't (and probably shouldn't be) run in parallel, so there's no need to rescope `tc`